src/createAgentManager.ts

```diff
@@ -38,18 +39,31 @@ function getAgentStreamArgs(agent: Agent): CreateStreamOptions {
     };
 }
 
-function initializeStreamAndChat(agent: Agent, options: AgentManagerOptions, agentsApi: AgentsAPI, chat?: Chat) {
+function initializeStreamAndChat(
+    agent: Agent,
+    options: AgentManagerOptions,
+    agentsApi: AgentsAPI,
+    analytics: AnalyticsProvider,
+    chat?: Chat
+) {
     return new Promise<{ chat: Chat; streamingManager: StreamingManager<ReturnType<typeof getAgentStreamArgs>> }>(
         async (resolve, reject) => {
             const streamingManager = await createStreamingManager(getAgentStreamArgs(agent), {
                 ...options,
+                analytics,
                 callbacks: {
                     ...options.callbacks,
                     onConnectionStateChange: async state => {
                         if (state === 'connected') {
                             try {
                                 if (!chat) {
                                     chat = await agentsApi.newChat(agent.id);
+                                    analytics.setChatId(chat.id);
+                                    analytics.track('agent-new-chat', {
+                                        event: 'sdk',
```

Comment:
event 'sdk' ?
should be created
then when closed it should be event: 'closed' / 'done'

Reply:
agent-chat, event: created **

Reply:
updated